distributed/formats/csv.py

```diff
@@ -0,0 +1,89 @@
+import pandas as pd
+from dask import do
+from dask.dataframe import from_imperative
+
+from ..executor import default_executor, ensure_default_get
+from ..utils import ensure_bytes
+
+from .compression import compressors, decompressors
+
+
+def bytes_read_csv(b, header, kwargs):
+    """ Convert a block of bytes to a Pandas DataFrame
+
+    Parameters
+    ----------
+    b: bytestring
+        The content to be parsed with pandas.read_csv
+    header: bytestring
+        An optional header to prepend to b
+    kwargs: dict
+        A dictionary of keyword arguments to be passed to pandas.read_csv
+
+    See Also:
+        distributed.formats.csv.read_csv
+    """
+    from io import BytesIO
+    import pandas as pd
+    with log_errors():
+        compression = kwargs.pop('compression', None)
+        b2 = decompressors[compression](b)
+        bio = BytesIO()
+        if header:
+            if not header.endswith(b'\n') and not header.endswith(b'\r'):
+                header = header + ensure_bytes(kwargs.get('lineterminator', b'\n'))
```

Comment:
It doesn't appear obvious to me that passing around the text header is the right way - shouldn't it be a list of column names (strings) or perhaps an empty dataframe?